### ISSUE TITLE
Non-redraw event dispatched durning redraw phase

### ISSUE THREAD

**Issue by pezcode** [rust-windowing/winit#1382]
I'm using the same redraw event pattern as shown [in the documention](https://docs.rs/winit/0.20.0/winit/#event-handling):

```
event_loop.run(move |event, _, control_flow| {
    *control_flow = ControlFlow::Poll;
    match event {
        Event::MainEventsCleared => {
                self.window.request_redraw();
        }
        Event::RedrawRequested(window_id) if window_id == self.window.id() => {
                self.context.draw_frame();
        }
    };
})
```

which works, but gives me the following warning:

> [winit::platform_impl::platform::event_loop::runner] Non-redraw event dispatched durning redraw phase

This is on Windows 10 (1909), Rust 1.40 + stable-x86_64-pc-windows-msvc

Removing the call to `request_redraw` removes the warning but then it won't redraw the window continuously. Requesting the redraw during `RedrawEventsCleared` instead also works but produces the same warning.

Am I doing something wrong here? The warning message sounds illogical seeing how all I'm dispatching is that redraw event.

**Comment by Osspial** [comment]
Thanks for reporting! Fortunately, this issue was fixed by #1366 - it just hasn't landed on master yet.

**Comment by aloucks** [comment]
@Osspial I'm still seeing this on the master branch, Windows 10.

I can't reproduce it 100% of the time, but it's most often after clicking into another window (focus lost) then clicking on the title bar of the winit window and attempting to move it. Sometimes it'll panic on the first try, other times it'll take 10+.

```rust
use winit::{
    event::{ElementState, Event, WindowEvent},
    event_loop::{ControlFlow, EventLoop},
    window::WindowBuilder,
};

use std::time::{Instant, Duration};

fn main() {
    simple_logger::init().unwrap();
    let event_loop = EventLoop::new();

    let window = WindowBuilder::new()
        .with_title("A fantastic window!")
        .build(&event_loop)
        .unwrap();

    event_loop.run(move |event, _, control_flow| {
        //println!("{:?}", event);

        *control_flow = ControlFlow::WaitUntil(Instant::now() + Duration::from_secs_f32(0.25));

        match event {
            Event::WindowEvent { event, .. } => match event {
                WindowEvent::CloseRequested => *control_flow = ControlFlow::Exit,
                _ => (),
            },
            Event::MainEventsCleared => {
                window.request_redraw();
            }
            Event::RedrawRequested(_) => {
                println!("redrawing! {:?}", Instant::now());
            }
            _ => (),
        }
    });
}
```
